src/rust/engine/src/session.rs

```diff
@@ -101,6 +101,9 @@ struct SessionHandle {
   // Whether or not this Session has been cancelled. If a Session has been cancelled, all work that
   // it started should attempt to exit in an orderly fashion.
   cancelled: AsyncLatch,
+  // True if this Session should be shielded from keyboard interrupts (which cancel all
+  // non-isolated Sessions).
+  isolated: bool,
```

Comment:
Can cancelled just be wrapped in an Option instead and then the cancel impl just map over that?

Reply:
Yea, it could. I was thinking though that being isolated/background didn't necessarily mean that you weren't cancellable at all: just that you shouldn't be canceled by SIGINT. It's speculative though.